tests/tools/MemoryIStream.cc

```diff
@@ -0,0 +1,39 @@
+#define CATCH_CONFIG_MAIN
+
+#include "third-party/Catch/single_include/catch2/catch.hpp"
+
+#include <cstring>
+#include <ratio>
+
+#include "tools/ContiguousStream.h"
+#include "tools/MemoryIStream.h"
+#include "tools/Random.h"
+
+TEST_CASE("MemoryIStream", "[tools]") {
+
+  const size_t num_ints = std::kilo{}.num;
+  emp::vector<int> source(num_ints * sizeof(int));
+
+  emp::Random rng{ 1 };
+  rng.RandFill(
+    reinterpret_cast<unsigned char*>( source.data() ),
```

Comment:
maybe we can use https://github.com/devosoft/Empirical/blob/bcb32f54fd9df4b1acf9c444dc3a8c5ec57c41de/source/tools/random_utils.h#L95 since it seems like a simpler interface